includes/controllers/class.llms.controller.orders.php

```diff
@@ -415,51 +416,75 @@ public function pending_cancel_order( $order ) {
 	 *
 	 * @since 3.0.0
 	 * @since 3.32.0 Record order notes and trigger actions during errors.
+	 * @since [version] Made sure to process only proper LLMS_Orders of existing users.
 	 *
 	 * @param int $order_id WP Post ID of the order.
 	 * @return void
 	 */
 	public function recurring_charge( $order_id ) {
 
-		$order   = new LLMS_Order( $order_id );
-		$gateway = $order->get_gateway();
+		$order = llms_get_post( $order_id );
 
-		// ensure the gateway is still installed & available
-		if ( ! is_wp_error( $gateway ) ) {
+		if ( ! ( $order && is_a( $order, 'LLMS_Order' ) ) ) {
```

Comment:
I think this is a confusing syntax. I'd rather ` if ( ! $order || ! is_a( $order, 'LLMS_Order  ) {`

Reply:
Mmm okay, I find it more easy for my brain: "I want both or nothing..." 😁